sources/Metal/Renderers/iTermCursorGuideRenderer.m

```diff
@@ -1,60 +1,96 @@
 #import "iTermCursorGuideRenderer.h"
 
 @interface iTermCursorGuideRendererTransientState()
-@property (nonatomic, strong) id<MTLTexture> texture;
+@property (nonatomic, strong) id<MTLTexture> horizontalTexture, verticalTexture;
 @property (nonatomic) int row;
 @property (nonatomic) int column;
+@property (nonatomic) id<MTLBuffer> horizontalVertexBuffer, verticalVertexBuffer, upperVertexBuffer, lowerVertexBuffer;
 @end
 
 @implementation iTermCursorGuideRendererTransientState {
     int _row;
     int _column;
+    id<MTLBuffer> _horizontalVertexBuffer, _verticalVertexBuffer;
+    id<MTLBuffer> _upperVertexBuffer, _lowerVertexBuffer;
 }
 
 - (void)setCursorCoord:(VT100GridCoord)coord within:(VT100GridSize)bounds {
     _row = (0 <= coord.y && coord.y < bounds.height) ? coord.y : -1;
     _column = (0 <= coord.x && coord.x < bounds.width)  ? coord.x : -1;
 }
 
-- (void)initializeVerticesWithPool:(iTermMetalBufferPool *)verticesPool {
+- (void)initializeVerticesWithPool:(iTermMetalBufferPool *)verticesPool
+                        horizontal:(BOOL)horizontal
+                          vertical:(BOOL)vertical {
```

Comment:
I suppose it would be an error to call this with both horizontal and vertical set to false, so add an assertion that at least one is set.

Reply:
Added assert.